factory_reset/factory_reset.py

```diff
@@ -0,0 +1,38 @@
+import time
+import RPi.GPIO as GPIO
+import os
+import subprocess  # nosec #pylint-disable type: ignore
+import alsaaudio
+
+
+current_folder = os.path.dirname(os.path.abspath(__file__))
+factory_reset = os.path.join(current_folder,'factory_reset.sh')
+try:
+    GPIO.setmode(GPIO.BCM)
+    GPIO.setup(17,GPIO.IN)
+    i = 1
+    while True:
+        if GPIO.input(17) == 1:
+            pass
```

Comment:
So, if the GPIO is still high (1), after reading, the next read will start right away?
This will make CPU tired. Put a short delay (`time.sleep`) here.